packages/api/lambdas/sf-event-sqs-to-db-records.js

```diff
@@ -1,22 +1,95 @@
+
 const { parseSQSMessageBody, sendSQSMessage } = require('@cumulus/aws-client/SQS');
+const StepFunctions = require('@cumulus/aws-client/StepFunctions');
 const log = require('@cumulus/common/log');
-const { getMessageExecutionArn } = require('@cumulus/message/Executions');
+const { getKnexClient } = require('@cumulus/db');
+const {
+  getMessageExecutionArn,
+  getMessageExecutionParentArn,
+} = require('@cumulus/message/Executions');
 const Execution = require('../models/executions');
 const Granule = require('../models/granules');
 const Pdr = require('../models/pdrs');
 const { getCumulusMessageFromExecutionEvent } = require('../lib/cwSfExecutionEventUtils');
 
-const saveExecutionToDb = async (cumulusMessage) => {
+const isPostRDSDeploymentExecution = (cumulusMessage) => {
+  const minimumSupportedRDSVersion = process.env.RDS_DEPLOYMENT_CUMULUS_VERSION;
+  if (!minimumSupportedRDSVersion) {
+    throw new Error('RDS_DEPLOYMENT_CUMULUS_VERSION environment variable must be set');
+  }
+  const cumulusVersion = get(cumulusMessage, 'cumulus_meta.cumulus_version', '0.0.0');
+  return semver.gte(cumulusVersion, minimumSupportedRDSVersion);
+};
+
+/**
+ * Get the initial execution message for a chain of workflows.
+ *
+ * @param {Object} cumulusMessage - A Cumulus workflow message
+ * @returns {Object}
+ *   Cumulus workflow message for the parent execution initiating
+ *   this chain of executions.
+ */
+const getInitialExecutionMessage = async (cumulusMessage) => {
+  const parentArn = getMessageExecutionParentArn(cumulusMessage);
+  if (!parentArn) {
+    return cumulusMessage;
+  }
+  const executionArn = getMessageExecutionArn(cumulusMessage);
+  const response = await StepFunctions.describeExecution({
+    executionArn,
+  });
+  return getInitialExecutionMessage(JSON.parse(response.input));
+};
+
+const shouldWriteToRDS = async (cumulusMessage) => {
+  const initialMessage = await getInitialExecutionMessage(cumulusMessage);
+  return isPostRDSDeploymentExecution(initialMessage);
+};
+
+const saveExecutionToDynamoDb = async (cumulusMessage, executionModel) =>
+  executionModel.storeExecutionFromCumulusMessage(cumulusMessage);
+
+const deleteExecutionFromRDS = async (isRDSWriteEnabled, executionArn, knex) => {
+  if (!isRDSWriteEnabled) return true;
+  return knex('executions')
+    .where({
+      arn: executionArn,
+    })
+    .delete();
+};
+
+// Just a stub for write functionality
+const saveExecutionToRDS = async (isRDSWriteEnabled, cumulusMessage, knex) => {
+  if (!isRDSWriteEnabled) return true;
+  const executionArn = getMessageExecutionArn(cumulusMessage);
+  return knex('executions').insert({
+    arn: executionArn,
+  });
+};
+
+const saveExecutions = async (cumulusMessage, isRDSWriteEnabled, knex) => {
   const executionModel = new Execution();
+  const executionArn = getMessageExecutionArn(cumulusMessage);
   try {
-    await executionModel.storeExecutionFromCumulusMessage(cumulusMessage);
+    await pAll([
```

Comment:
Is there a reason this is not just using Promise.allSettled with a filter?

Reply:
(that's not to suggest it *needs* one, just curious)

Reply:
No, there's not. It's probably better to choose one approach or the other. I guess I just found pAll a little easier to use/understand, but Promise.allSettled being natively supported probably pushes the needle that direction